in_toto/runlib.py

```diff
@@ -137,18 +138,29 @@ def record_artifacts_as_dict(artifacts):
     original_cwd = os.getcwd()
     try:
       os.chdir(in_toto.settings.ARTIFACT_BASE_PATH)
-    except OSError as e:
-      raise OSError("Review your ARTIFACT_BASE_PATH setting - {}".format(e))
+    except Exception as e:
+      raise in_toto.exceptions.SettingsError(
+          "Review your ARTIFACT_BASE_PATH setting '{}' - {}".format(
+          in_toto.settings.ARTIFACT_BASE_PATH, e))
 
   # Normalize passed paths
   norm_artifacts = []
   for path in artifacts:
     norm_artifacts.append(os.path.normpath(path))
 
+  # If ARTIFACT_EXCLUDES is set it must be a list of strings or an empty
+  # TODO: Change NAMES_SCHEMA to something more semantically accurate
+  if (in_toto.settings.ARTIFACT_EXCLUDES and not
+      securesystemslib.formats.NAMES_SCHEMA.matches(
```

Comment:
Maybe PATH_SCHEMA or PATHS_SCHEMA is a bit more accurate?  I am not certain, but ARTIFACT_EXCLUDES appears to be glob patterns?